### ISSUE TITLE
Coroutines variant of Panache Repository

### ISSUE THREAD

**Issue by hantsy** [quarkusio/quarkus#34624]
### Description

To use the Repository pattern and Kotlin Coroutines, I would like there is Coroutines variant and return Coroutines flavor result directly(no transform in the codes).

```kotlin 
interface CoroutinePanacheRepositoryBase<E, ID> {

   suspend findById(ID): T?
   findAll():Flow<E>
}
```

In Spring and Micronaut, the basic Kotlin support just uses some Kotlin extensions to extend the functionality on the existing Java API, only Coroutines used standalone API for it.

**Comment by geoand** [comment]
I am pretty sure we already have an issue for this, but I'm on a phone now so it's hard to check

**Comment by geoand** [comment]
We already have this in the `quarkus-hibernate-reactive-panache-kotlin`

**Comment by hantsy** [comment]
@geoand I do not find this in mongo kotlin support. https://github.com/hantsy/quarkus-sandbox/blob/master/mongodb-kotlin-co/src/main/kotlin/com/example/PostRepository.kt

**Comment by geoand** [comment]
The issue never mentioned Mongo to begin with, but yeah, good point.

I'll reopen the issue and fix the title.

**Comment by hantsy** [comment]
@geoand Why the common Repository interfaces are not shared in all data storage solutions? like Spring Data and Micronaut Data.

**Comment by hantsy** [comment]
> We already have this in the `quarkus-hibernate-reactive-panache-kotlin`

@geoand  I can not find it here, https://github.com/quarkusio/quarkus/tree/3.2.2.Final/extensions/panache/hibernate-reactive-panache-kotlin/runtime/src/main/kotlin/io/quarkus/hibernate/reactive/panache/kotlin

**Comment by geoand** [comment]
You are right... I thought I saw some Kotlin suspend functions, but I can't find them know... @evanchooly what's the status of Hibernate Reactive + Kotlin Coroutines?

**Comment by evanchooly** [comment]
I don't know that any work has been done on HR and kotlin.  HR was pretty new last time I did any such work and I just haven't gotten around to it.  Reactive isn't really my forte so I'd have to do a bit of research.  It's probably as much as work as the others, though, so we have a pretty solid game plan for such things.

**Comment by hantsy** [comment]
@geoand BTW, why we need a kotlin variant of [PanacheRepository](https://github.com/quarkusio/quarkus/blob/3.2.2.Final/extensions/panache/hibernate-reactive-panache-kotlin/runtime/src/main/kotlin/io/quarkus/hibernate/reactive/panache/kotlin/PanacheRepository.kt), in Kotlin codes, we can use the Java interface directly. And in Spring Data/Micronaut Data, this case, only adding some extension functions to these interfaces.

**Comment by kuoche1712003** [comment]
You can [awaiting a Uni in coroutines](https://smallrye.io/smallrye-mutiny/2.4.0/guides/kotlin/#awaiting-a-uni-in-coroutines).
Maybe you don't need a kotlin variant of PanacheRepository.
